docs/dotnet/dotnet-aspnetcore-metrics.md

```diff
@@ -0,0 +1,226 @@
+
+**[1]:** MUST NOT be populated when this is not supported by the HTTP server framework as the route attribute should have low-cardinality and the URI path can NOT substitute it.
+SHOULD include the [application root](/docs/http/http-spans.md#http-server-definitions) if there is one.
+
+`aspnetcore.routing.match_status` has the following list of well-known values. If one of them applies, then the respective value MUST be used, otherwise a custom value MAY be used.
+
+| Value  | Description |
+|---|---|
+| `success` | Match succeeded |
+| `failure` | Match failed |
+<!-- endsemconv -->
+
+## Exception metrics
+
+Metrics reported by `Microsoft.AspNetCore.Diagnostics` meter.
+
+### Metric: `aspnetcore.diagnostics.exceptions`
+
+<!-- semconv metric.aspnetcore.diagnostics.exceptions(metric_table) -->
+| Name     | Instrument Type | Unit (UCUM) | Description    |
+| -------- | --------------- | ----------- | -------------- |
+| `aspnetcore.diagnostics.exceptions` | Counter | `{exception}` | Number of exceptions caught by exception handling middleware. [1] |
+
+**[1]:** Meter name: `Microsoft.AspNetCore.Diagnostics`; Added in: ASP.NET Core 8.0
+<!-- endsemconv -->
+
+<!-- semconv metric.aspnetcore.diagnostics.exceptions(full) -->
+| Attribute  | Type | Description  | Examples  | Requirement Level |
+|---|---|---|---|---|
+| `aspnetcore.diagnostics.exception.result` | string | ASP.NET Core exception middleware handling result | `handled`; `unhandled` | Required |
+| `aspnetcore.diagnostics.handler.type` | string | Full type name of the [`IExceptionHandler`](https://learn.microsoft.com/dotnet/api/microsoft.aspnetcore.diagnostics.iexceptionhandler) implementation that handled the exception. | `Contoso.MyHandler` | Conditionally Required: [1] |
+| `error.type` | string | The full name of exception type. [2] | `System.OperationCanceledException`; `Contoso.MyException` | Required |
+
+**[1]:** if and only if the exception was handled by this handler.
+
+**[2]:** The `error.type` SHOULD be predictable and SHOULD have low cardinality.
+Instrumentations SHOULD document the list of errors they report.
+
+The cardinality of `error.type` within one instrumentation library SHOULD be low, but
+telemetry consumers that aggregate data from multiple instrumentation libraries and applications
+should be prepared for `error.type` to have high cardinality at query time, when no
+additional filters are applied.
+
+If the operation has completed successfully, instrumentations SHOULD NOT set `error.type`.
+
+If a specific domain defines its own set of error identifiers (such as HTTP or gRPC status codes),
+it's RECOMMENDED to use a domain-specific attribute and also set `error.type` to capture
+all errors, regardless of whether they are defined within the domain-specific set or not.
```

Comment:
Did you decide to leave the all-caps instances unchanged?

Reply:
> Did you decide to leave the all-caps instances unchanged?

Yes, that's the style at https://opentelemetry.io/docs/specs/otel/metrics/semantic_conventions/
If I get approval from opentelemetry, I'll change all their docs, then change ours. I've got a PR there they like, but it's not that aggressive and no CAPS STYLE changes.  see https://github.com/open-telemetry/semantic-conventions/pull/188
I need to close that PR and make my changes in the correct files since the doc's are generated from YAML.

Reply:
As a heads up, we also have a version of this aimed at the learn.microsoft docs: https://github.com/dotnet/docs/pull/37213. The main differences were:
- Converting formal specification language to plainer English (such as removing the caps)
- Reframing the text as a description of what .NET actually implemented as opposed to a specification describing what is permitted or recommended to implement.
- Reduced amount of minutiae (we can always point back to these docs for that)
- Cross-linked to other relevant learn.microsoft docs